File: android/src/main/java/com/lange/bluetooth_printer/BluetoothPrinterPlugin.java
package com.lange.bluetooth_printer;

import com.google.gson.Gson;

import io.flutter.plugin.common.EventChannel;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/**
 * BluetoothPrinterPlugin
 */
public class BluetoothPrinterPlugin implements MethodCallHandler {
    /**
     * Plugin registration.
     */
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "bluetooth_printer/methodChannel");
        channel.setMethodCallHandler(new BluetoothPrinterPlugin());

        EventChannel eventChannel = new EventChannel(registrar.messenger(), "bluetooth_printer/scanBlueToothEvent");
        ScanBlueToothEventListener scanBlueToothEventListener = new ScanBlueToothEventListener(registrar.activity());
        eventChannel.setStreamHandler(scanBlueToothEventListener);
    }

    BlueToothPrinter blueToothPrinter;

    @Override
    public void onMethodCall(MethodCall call, Result result) {
        if(blueToothPrinter==null){
            blueToothPrinter = new BlueToothPrinter();
        }
        if (call.method.equals("startScanBlueTooth")) {
            blueToothPrinter.startScanBlueTooth();
        } else if(call.method.equals("isConnected")){
            boolean b = blueToothPrinter.isConnected();
            result.success(b?1:0);
        }else if(call.method.equals("connectBlueTooth")){
            int index = call.argument("index");
            blueToothPrinter.connectBlueTooth(index);
            result.success(1);
        }else if(call.method.equals("print")){
            String orderJsonStr = call.argument("orderJsonStr");
            OrderModel orderModel = new Gson().fromJson(orderJsonStr,OrderModel.class);
            blueToothPrinter.print(orderModel);
        }

        else {
            result.notImplemented();
        }
    }
}

File: ios/Classes/BluetoothPrinterPlugin.m
#import "BluetoothPrinterPlugin.h"
#import "BlueToothPrinter.h"
#import "ScanBlueToothEvent.h"
@implementation BluetoothPrinterPlugin{
     BlueToothPrinter *blueToothPrinter;
}
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    /*方法通道*/
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:@"bluetooth_printer/methodChannel"
                                     binaryMessenger:[registrar messenger]];
    BluetoothPrinterPlugin* instance = [[BluetoothPrinterPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
    /*事件通道*/
    FlutterEventChannel* eventChannle = [FlutterEventChannel eventChannelWithName:@"bluetooth_printer/scanBlueToothEvent" binaryMessenger:[registrar messenger]];
    ScanBlueToothEvent *scanBlueToothEvent=[[ScanBlueToothEvent alloc]init];
    [eventChannle setStreamHandler:scanBlueToothEvent];

}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    if(!blueToothPrinter){
        blueToothPrinter =[[BlueToothPrinter alloc]init];
    }
    if([@"startScanBlueTooth" isEqualToString:call.method]){
        [blueToothPrinter startScanBlueTooth];
    }
    else if([@"connectBlueTooth" isEqualToString:call.method]){
        int index = [call.arguments[@"index"] intValue];
        [blueToothPrinter connectBlueTooth:index];
        result(@(1));
    }else if([@"print" isEqualToString:call.method]){
        NSString * orderJsonStr = call.arguments[@"orderJsonStr"];
        [blueToothPrinter print:orderJsonStr];
    }
    else if([@"isConnected" isEqualToString:call.method]){
        int r = [blueToothPrinter isConnected];
        result(@(r));
    }
    else {
        result(FlutterMethodNotImplemented);
    }
}

@end

